webpack.renderer.ts

```diff
@@ -1,17 +1,38 @@
-import { appName, htmlTemplate, isDevelopment, isProduction, buildDir, rendererDir, sassCommonVars, publicPath } from "./src/common/vars";
+import { appName, buildDir, extensionsDir, extensionsLibName, htmlTemplate, isDevelopment, isProduction, publicPath, rendererDir, sassCommonVars } from "./src/common/vars";
 import path from "path";
 import webpack from "webpack";
 import HtmlWebpackPlugin from "html-webpack-plugin";
 import MiniCssExtractPlugin from "mini-css-extract-plugin";
 import TerserPlugin from "terser-webpack-plugin";
 import ForkTsCheckerPlugin from "fork-ts-checker-webpack-plugin"
 
-export default function (): webpack.Configuration {
-  console.info('WEBPACK:renderer', require("./src/common/vars"))
+export default [
+  webpackLensRenderer,
+  webpackExtensionsApi,
+]
+
+// todo: use common chunks/externals for "react", "react-dom", etc.
+export function webpackExtensionsApi(): webpack.Configuration {
+  const config = webpackLensRenderer({ showVars: false });
+  config.name = "extensions-api"
+  config.entry = {
+    [extensionsLibName]: path.resolve(extensionsDir, "extension-api.ts")
+  };
+  config.output.libraryTarget = "commonjs2"
+  delete config.devtool;
+  delete config.plugins;
+  return config;
+}
+
+export function webpackLensRenderer({ showVars = true } = {}): webpack.Configuration {
+  if (showVars) {
+    console.info('WEBPACK:renderer', require("./src/common/vars"));
+  }
   return {
     context: __dirname,
     target: "electron-renderer",
     devtool: "source-map", // todo: optimize in dev-mode with webpack.SourceMapDevToolPlugin
+    name: "lens-app",
```

Comment:
where do we use this?

Reply:
This is visible during compilation, e.g.
```
Hash: 576d90d865c308e18eea
Version: webpack 4.43.0
Child **lens-app**:
    Hash: 576d90d865c308e18eea
    Time: 5913ms
```

Also you can run `webpack --name lens-app` to compile single configuration in this file.

Reply:
thanks. So not visible to end user while using the app?

Reply:
yes, this is totally hidden in UI :)